Sources/DatadogSDKTesting/DDTestMonitor.swift

```diff
@@ -73,6 +76,31 @@ internal class DDTestMonitor {
                         DDCrashes.setCustomData(customData: SimpleSpanSerializer.serializeSpan(simpleSpan: simpleSpan))
                     }
                 }
+
+            didBecomeActiveNotificationObserver = NotificationCenter.default.addObserver(
+                forName: didBecomeActiveNotificationName,
+                object: nil, queue: nil) { _ in
+                    var data = [DDUISettingsTags.uiSettingsAppearance: PlatformUtils.getAppearance(),
+                                DDUISettingsTags.uiSettingsLocalization: PlatformUtils.getLocalization()]
+                    #if os(iOS)
+                    data[DDUISettingsTags.uiSettingsOrientation] = PlatformUtils.getOrientation()
+                    #endif
+                    let encoded = try? JSONSerialization.data(withJSONObject: data)
+                    let timeout: CFTimeInterval = 1.0
+                    let remotePort = CFMessagePortCreateRemote(nil, "DatadogTestingPort" as CFString)
+                    let status = CFMessagePortSendRequest(remotePort,
+                                                          0x1111, // Arbitrary
+                                                          encoded as CFData?,
+                                                          timeout,
+                                                          timeout,
+                                                          nil,
+                                                          nil)
```

Comment:
This is really cool! Where do we receive this message? Can this be easily used as general IPC solution to communicate "app under test" and "UI Test runner" processes? Can it be bidirectional communication?

Reply:
My tests showed that it was working at least on the simulator, so I would say that it can be used as a general solution. It looks like sandboxing problems don't arise between the test runner and the app, Xcode uses its own XPC services to connect between them so it must be solving the issue for us. I didn't check the bidirectional communication, but I don't see a reason why it would not work

Reply:
OK, so it's using the macOS capabilities for Simulator. I wonder it this will work in UI tests run on a physical device, any idea?

Reply:
No, it does not, CFMessagePortCreateLocal returns nil, so it is not usable on physical devices